Source/WebCore/css/parser/CSSParserImpl.cpp

```diff
@@ -558,20 +558,66 @@ RefPtr<StyleRuleNamespace> CSSParserImpl::consumeNamespaceRule(CSSParserTokenRan
     return StyleRuleNamespace::create(namespacePrefix, uri);
 }
 
-RefPtr<StyleRuleMedia> CSSParserImpl::consumeMediaRule(CSSParserTokenRange prelude, CSSParserTokenRange block)
+void CSSParserImpl::executeInNewStackContext(auto&& run)
+{
+    // Save the current stack context
+    auto parsedRules = WTFMove(m_parsedRules);
+    auto parsedProperties = WTFMove(m_parsedProperties);
+    m_parsedRules.clear();
+    m_parsedProperties.clear();
+
+    run();
+
+    // Restore the old stack context
+    m_parsedRules = parsedRules;
+    m_parsedProperties = parsedProperties;
+}
```

Comment:
An alternative strategy would be to have something like

```
struct NestingContext {
   Vector<Ref<StyleRuleBase>> parsedRules;
   ParsedPropertyVector parsedProperties
};
Vector<NestingContext> m_nestingContextStack;
```

and push/pop the stack. The code would always use whatever is on top of the stack (`m_nestingContextStack.last().parsedRules` etc). You'd push once to always have a top level context.

This has the additional advantage that you don't need a separate m_styleRuleNestingDepth field. The stack depth is your nesting depth.